src/app/loadout-drawer/loadout-drawer-reducer.ts

```diff
@@ -676,3 +669,163 @@ export function removeArtifactUnlock(mod: number): LoadoutUpdateFunction {
     }
   });
 }
+
+/** Replace the loadout's subclass with the store's currently equipped subclass */
+export function randomizeLoadoutSubclass(
+  defs: D1ManifestDefinitions | D2ManifestDefinitions,
+  store: DimStore
+): LoadoutUpdateFunction {
+  return (loadout) => {
+    const newSubclass = _.sample(
+      store.items.filter(
+        (item) => item.bucket.hash === BucketHashes.Subclass && itemCanBeInLoadout(item)
+      )
+    );
+
+    if (!newSubclass || !defs.isDestiny2()) {
+      return loadout;
+    }
+
+    const newLoadoutItem: LoadoutItem = {
+      ...convertToLoadoutItem(newSubclass, true),
+      socketOverrides: randomSubclassConfiguration(defs, newSubclass),
+    };
+
+    const isSubclass = (i: LoadoutItem) =>
+      defs.InventoryItem.get(i.hash)?.inventory?.bucketTypeHash === BucketHashes.Subclass;
+
+    const newLoadout = {
+      ...loadout,
+      items: [...loadout.items.filter((i) => !isSubclass(i)), newLoadoutItem],
+    };
```

Comment:
I *think* `addItem` will do the right thing if you just try to add a new subclass - if not, it'd be helpful to have a "setSubclass" helper.